code/tests/functional/backend_api/test_azure_byod.py

```diff
@@ -0,0 +1,128 @@
+
+pytestmark = pytest.mark.functional
+
+path = "/api/conversation/azure_byod"
+body = {
+    "conversation_id": "123",
+    "messages": [
+        {"role": "user", "content": "Hello"},
+        {"role": "assistant", "content": "Hi, how can I help?"},
+        {"role": "user", "content": "What is the meaning of life?"},
+    ],
+}
+
+
+def test_azure_byod_responds_successfully_when_streaming(
+    app_url: str, app_config: AppConfig, httpserver: HTTPServer
+):
+    # given
+    httpserver.clear_all_handlers()  # Clear default successful responses
+
+    httpserver.expect_request(
+        f"/openai/deployments/{app_config.get('AZURE_OPENAI_MODEL')}/chat/completions",
+        method="POST",
+    ).respond_with_data(
+        Template(
+            r"""data: {"id":"92f715be-cfc4-4ae6-80f8-c86b7955f6af","model":"$model","created":1712077271,"object":"extensions.chat.completion.chunk","choices":[{"index":0,"delta":{"role":"assistant","context":{"citations":[{"content":"document","title":"/documents/doc.pdf","url":null,"filepath":null,"chunk_id":"0"}],"intent":"[\"intent\"]"}},"end_turn":false,"finish_reason":null}]}
+
+data: {"id":"92f715be-cfc4-4ae6-80f8-c86b7955f6af","model":"$model","created":1712077271,"object":"extensions.chat.completion.chunk","choices":[{"index":0,"delta":{"content":"42 is the meaning of life"},"end_turn":false,"finish_reason":null}],"system_fingerprint":"fp_68a7d165bf"}
+
+data: {"id":"92f715be-cfc4-4ae6-80f8-c86b7955f6af","model":"$model","created":1712077271,"object":"extensio@ns.chat.completion.chunk","choices":[{"index":0,"delta":{},"end_turn":true,"finish_reason":"stop"}]}
+
+data: [DONE]
+"""
+        ).substitute(model=app_config.get("AZURE_OPENAI_MODEL"))
+    )
+
+    # when
+    response = requests.post(f"{app_url}{path}", json=body)
+
+    # then
+    assert response.status_code == 200
+
+    response_lines = response.text.splitlines()
+    assert len(response_lines) == 3
+
+    final_response_json = json.loads(response_lines[-1])
+    assert final_response_json == {
+        "id": "92f715be-cfc4-4ae6-80f8-c86b7955f6af",
+        "model": app_config.get("AZURE_OPENAI_MODEL"),
+        "created": 1712077271,
+        "object": "extensions.chat.completion.chunk",
+        "choices": [
+            {
+                "messages": [
+                    {
+                        "content": r'{"citations": [{"content": "document", "title": "/documents/doc.pdf", "url": null, "filepath": null, "chunk_id": "0"}], "intent": "[\"intent\"]"}',
+                        "end_turn": False,
+                        "role": "tool",
+                    },
+                    {
+                        "content": "42 is the meaning of life",
+                        "end_turn": True,
+                        "role": "assistant",
+                    },
+                ]
+            }
+        ],
+    }
+
+    assert response.headers["Content-Type"] == "application/json-lines"
+
+    verify_request_made(
```

Comment:
This test is quite large, is it worth pulling this verification out into it's own test to break it up and reduce the number of reasons this test could fail? Same for the without data test.